diff --git a/dist/index.cjs b/dist/index.cjs
index 7aad527fc317f6ce42d6520b54bf7b63f0541681..5b0cce020d0ebe22cb22d4ce86ee92c7a190fd81 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -13066,7 +13066,7 @@ async function getMediaUrl(version, IdMedia, numberId, Token) {
     }
 }
 
-const processIncomingMessage = async ({ messageId, messageTimestamp, pushName, message, to, jwtToken, version, numberId, fileData }) => {
+const processIncomingMessage = async ({ messageId, messageTimestamp, pushName, message, to, jwtToken, version, numberId, fileData, quoted }) => {
     let responseObj;
     switch (message.type) {
         case 'text': {
@@ -13077,6 +13077,7 @@ const processIncomingMessage = async ({ messageId, messageTimestamp, pushName, m
                 body: message.text?.body,
                 name: pushName,
                 pushName,
+                quoted,
             };
             break;
         }
@@ -13095,6 +13096,7 @@ const processIncomingMessage = async ({ messageId, messageTimestamp, pushName, m
                     : undefined,
                 pushName,
                 name: pushName,
+                quoted,
             };
             break;
         }
@@ -13108,6 +13110,7 @@ const processIncomingMessage = async ({ messageId, messageTimestamp, pushName, m
                 title_button_reply: message.button?.payload,
                 pushName,
                 name: pushName,
+                quoted,
             };
             break;
         }
@@ -13123,6 +13126,7 @@ const processIncomingMessage = async ({ messageId, messageTimestamp, pushName, m
                 body: bot.utils.generateRefProvider('_event_media_'),
                 pushName,
                 name: pushName,
+                quoted,
             };
             break;
         }
@@ -13137,6 +13141,7 @@ const processIncomingMessage = async ({ messageId, messageTimestamp, pushName, m
                 body: bot.utils.generateRefProvider('_event_document_'),
                 pushName,
                 name: pushName,
+                quoted,
             };
             break;
         }
@@ -13152,6 +13157,7 @@ const processIncomingMessage = async ({ messageId, messageTimestamp, pushName, m
                 body: bot.utils.generateRefProvider('_event_media_'),
                 pushName,
                 name: pushName,
+                quoted,
             };
             break;
         }
@@ -13165,6 +13171,7 @@ const processIncomingMessage = async ({ messageId, messageTimestamp, pushName, m
                 body: bot.utils.generateRefProvider('_event_location_'),
                 pushName,
                 name: pushName,
+                quoted,
             };
             break;
         }
@@ -13179,6 +13186,7 @@ const processIncomingMessage = async ({ messageId, messageTimestamp, pushName, m
                 body: bot.utils.generateRefProvider('_event_voice_note_'),
                 pushName,
                 name: pushName,
+                quoted,
             };
             break;
         }
@@ -13191,6 +13199,7 @@ const processIncomingMessage = async ({ messageId, messageTimestamp, pushName, m
                 body: bot.utils.generateRefProvider('_event_media_'),
                 pushName,
                 name: pushName,
+                quoted,
             };
             break;
         }
@@ -13208,6 +13217,7 @@ const processIncomingMessage = async ({ messageId, messageTimestamp, pushName, m
                 body: bot.utils.generateRefProvider('_event_contacts_'),
                 pushName,
                 name: pushName,
+                quoted,
             };
             break;
         }
@@ -13223,6 +13233,7 @@ const processIncomingMessage = async ({ messageId, messageTimestamp, pushName, m
                 body: bot.utils.generateRefProvider('_event_order_'),
                 pushName,
                 name: pushName,
+                quoted,
             };
             break;
         }
@@ -13310,6 +13321,7 @@ class MetaCoreVendor extends EventEmitter {
                         [contact] = contacts;
                     const to = body.entry[0].changes[0].value?.metadata?.display_phone_number;
                     const pushName = contact?.profile?.name ?? 'Unknown';
+					const quoted = message?.context?.id ?? null;
                     const fileData = message?.audio ??
                         message?.image ??
                         message?.video ??
@@ -13322,6 +13334,7 @@ class MetaCoreVendor extends EventEmitter {
                         to,
                         pushName,
                         message,
+						quoted,
                         jwtToken,
                         numberId,
                         version,
@@ -13920,6 +13933,24 @@ class MetaProvider extends bot.ProviderClass {
                 body.context = { message_id: context };
             return this.sendMessageMeta(body);
         };
+		this.sendMediaUrl = async (to, type, url, caption = '', filename = '') => {
+			const mediaContent = {
+				link: url,
+				caption: caption,
+			}
+			if (filename) {
+				mediaContent.filename = filename;
+			}
+			to = parseMetaNumber(to);
+			const body = {
+				messaging_product: 'whatsapp',
+				recipient_type: 'individual',
+				to,
+				type: type,
+				[type]: mediaContent,
+			}
+			return this.sendMessageMeta(body);
+		};
         this.sendReaction = async (to, react) => {
             to = parseMetaNumber(to);
             const body = {
@@ -13988,7 +14019,7 @@ class MetaProvider extends bot.ProviderClass {
             };
             return this.sendMessageMeta(body);
         };
-        this.sendText = async (to, message, context = null, preview_url = false) => {
+        this.sendText = async (to, message, context = null, preview_url = true) => {
             to = parseMetaNumber(to);
             const body = {
                 messaging_product: 'whatsapp',
